internal/overlord/servstate/handlers.go

```diff
@@ -126,14 +141,22 @@ func (m *ServiceManager) doStart(task *state.Task, tomb *tomb.Tomb) error {
 	}
 
 	active := &activeService{
-		cmd:       cmd,
-		done:      make(chan struct{}),
-		logBuffer: logBuffer,
+		plannedService: service.Copy(),
+		cmd:            cmd,
+		done:           make(chan struct{}),
+		logBuffer:      logBuffer,
 	}
 	m.services[req.Name] = active
 	go func() {
 		active.err = cmd.Wait()
 		_ = active.logBuffer.Close()
+		releasePlan, err := m.acquirePlan()
+		if err == nil {
+			if m.services[req.Name].cmd == cmd {
+				delete(m.services, req.Name)
+			}
+			releasePlan()
+		}
```

Comment:
But that change means we are now losing the metadata about the command termination. With the previous logic, we can comfortable look at the metadata for the running state at any point, while with this change it will be gone as soon as the service itself dies, so we in fact can never trust the cmd data to be there anymore because we don't control the process termination. Am I missing something?  If not and there wasn't a clear motivation for this change, I suggest reverting it for the time being.